tests/integration/suite/sentry/sentry.go

```diff
@@ -0,0 +1,406 @@
+			ctx, cancel := context.WithTimeout(parentCtx, 10*time.Second)
+			defer cancel()
+			conn, err := grpc.DialContext(
+				ctx,
+				fmt.Sprintf("127.0.0.1:%d", m.sentryWithJWKS.Port()),
+				grpc.WithTransportCredentials(transportCredentials),
+				grpc.WithReturnConnectionError(),
+				grpc.WithBlock(),
+			)
+			require.NoError(t, err)
+
+			client = sentrypbv1.NewCAClient(conn)
+		})
+
+		t.Run("fails when passing an invalid validator", func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(parentCtx, 10*time.Second)
+			defer cancel()
+			_, err := client.SignCertificate(ctx, &sentrypbv1.SignCertificateRequest{
+				Id:             defaultAppID,
+				Namespace:      defaultNamespace,
+				TokenValidator: sentrypbv1.SignCertificateRequest_TokenValidator(-1), // -1 is an invalid enum value
+			})
+			require.Error(t, err)
+
+			grpcStatus, ok := status.FromError(err)
+			require.True(t, ok)
+			require.Equal(t, codes.InvalidArgument, grpcStatus.Code())
+			require.Contains(t, grpcStatus.Message(), "not enabled")
+		})
+
+		t.Run("fails when passing the insecure validator", func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(parentCtx, 10*time.Second)
+			defer cancel()
+			_, err := client.SignCertificate(ctx, &sentrypbv1.SignCertificateRequest{
+				Id:             defaultAppID,
+				Namespace:      defaultNamespace,
+				TokenValidator: sentrypbv1.SignCertificateRequest_INSECURE,
+			})
+			require.Error(t, err)
+
+			grpcStatus, ok := status.FromError(err)
+			require.True(t, ok)
+			require.Equal(t, codes.InvalidArgument, grpcStatus.Code())
+			require.Contains(t, grpcStatus.Message(), "not enabled")
+		})
+
+		t.Run("fails when no validator is passed", func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(parentCtx, 10*time.Second)
+			defer cancel()
+			_, err := client.SignCertificate(ctx, &sentrypbv1.SignCertificateRequest{
+				Id:        defaultAppID,
+				Namespace: defaultNamespace,
+			})
+			require.Error(t, err)
+
+			grpcStatus, ok := status.FromError(err)
+			require.True(t, ok)
+			require.Equal(t, codes.InvalidArgument, grpcStatus.Code())
+			require.Contains(t, grpcStatus.Message(), "a validator name must be specified in this environment")
+		})
+
+		t.Run("issue a certificate with JWKS validator", func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(parentCtx, 10*time.Second)
+			defer cancel()
+
+			token, err := signJWT(generateJWT(fmt.Sprintf("spiffe://public/ns/%s/%s", defaultNamespace, defaultAppID)))
+			require.NoError(t, err)
+
+			res, err := client.SignCertificate(ctx, &sentrypbv1.SignCertificateRequest{
+				Id:                        defaultAppID,
+				Namespace:                 defaultNamespace,
+				CertificateSigningRequest: defaultAppCSR,
+				TokenValidator:            sentrypbv1.SignCertificateRequest_JWKS,
+				Token:                     string(token),
+			})
+			require.NoError(t, err)
+			require.NotEmpty(t, res.WorkloadCertificate)
```

Comment:
Again re checking the integrity and contents of the certificate.

Reply:
Done